website/docs/commands/publishing/publish.md

```diff
@@ -0,0 +1,243 @@
+your GPG key ring, or from external devices that GPG may support.
+
+To enable signing with GPG, pass `--gpg-key *key_id*` on the command line. If needed,
+you can specify arguments meant to be passed to `gpg`, with `--gpg-option`, like
+```text
+--gpg-key 1234567890ABCDEF --gpg-option --foo --gpg-option --bar
+```
+
+### Checksums
+
+Scala CLI can generate checksums of the artifacts it publishes.
+
+By default, Scala CLI generates SHA-1 and MD5 checksums. To disable checksums,
+pass `--checksum none`. To generate checksum formats to generate, pass them via
+`--checksum`, separating the checksum values with `,` or using `--checksum` multiple
+times:
+```text
+--checksum sha1,md5
+--checksum sha1 --checksum md5
+```
+
+To list supported checksum types, pass `--checksum list`.
+
+### CI overrides
+
+Scala CLI allows some publishing-related settings to have different values on your local machine and
+on CIs. In particular, this can be convenient to handle credentials and signing parameters, as these can
+be read from different locations on developers' machines and on CIs.
+
+On CIs (when `CI` is set in the environment, whatever its value), the CI override is
+used if it's there. Else the main directive is used.
+
+<!-- TODO Automatically generate that? -->
+| Settings | Directive | CI override directive |
+|----------|-------------------|-----------------------|
+| Compute Version | `publish.computeVersion` | `publish.ci.computeVersion` |
+| Repository | `publish.repository` | `publish.ci.repository` |
+| Repository User | `publish.user` | `publish.ci.user` |
+| Repository Password | `publish.password` | `publish.ci.password` |
+| Repository Realm | `publish.realm` | `publish.ci.realm` |
+| Secret Key | `publish.secretKey` | `publish.ci.secretKey` |
+| Secret Key Password | `publish.secretKeyPassword` | `publish.ci.secretKeyPassword` |
+| GPG key | `publish.gpgKey` | `publish.ci.gpgKey` |
+| GPG options | `publish.gpgOptions` | `publish.ci.gpgOptions` |
+
+## Repositories
+
+### Maven Central
+
+Use `central` as repository to push artifacts to Maven Central via `oss.sonatype.org`.
+To push to it via `s01.oss.sonatype.org` (the default for newly created repositories),
+use `central-s01`.
+
+### GitHub Packages
+
+Use `github` (GitHub organization and name computed from the git remotes)
+or `github:org/name` (replace `org` and `name` by the GitHub organization and name
+of your repository, like `github:VirtusLab/scala-cli`)
+to push artifacts to GitHub Packages.
+
+Note that, as of writing this, this disables parallel uploading of artifacts,
+checksums, and signing (all not supported by GitHub Packages as of writing this).
+
+### Ivy2 Local
+
+Use `ivy2Local` to put artifacts in the local Ivy2 repository, just like how
+[`publish local`](./publish-local.md) does.
+
+### Other pre-defined repositories
+
+All pre-defined repositories accepted by coursier, such as `jitpack` or `sonatype:snapshots`, are accepted as repositories for publishing.
```

Comment:
Does Coursier have a piece of doc about these repositories? If so, can we link it?